.github/workflows/azure-site-extension.yml

```diff
@@ -36,10 +36,39 @@ jobs:
           node-version: ${{ matrix.node-version }}
           architecture: ${{ matrix.arch }}
 
-      - name: Find agent version
+      - name: Get local agent version 
         run: |
+          $env:local_agent_version = node -p "require('./package.json').version"
+          echo "AGENT_VERSION=$env:local_agent_version" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+      - name: Find agent version at npm 
+        run: |
+          $env:npm_agent_version = npm view newrelic version 
+          echo "NPM_AGENT_VERSION=$env:npm_agent_version" | Out-File -FilePath $env:GITHUB_ENV -Append
+          
+      - name: Check NPM availability
+        if: ${{ env.NPM_AGENT_VERSION != env.AGENT_VERSION }}
+        run: |
+          $count = 0
+          while($count -lt 10) {
+            Start-Sleep -s 120
+            $npmversion = npm view newrelic version
+            echo "Checking npm ($count): $npmversion"
+            $test = [Version]$npmversion -match [Version]${{ env.AGENT_VERSION }}
+            if ([Version]$npmversion -match [Version]${{ env.AGENT_VERSION }}) {
+              break;
+            }
+            $count++
+          }
           $env:npm_agent_version = npm view newrelic version 
-          echo "AGENT_VERSION=$env:npm_agent_version" | Out-File -FilePath $env:GITHUB_ENV -Append
+          echo "Done with delayed check. Published version: $env:npm_agent_version Local version: ${{env.AGENT_VERSION}}"
+          echo "NPM_AGENT_VERSION=$env:npm_agent_version" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+      - name: Has the new agent been published?
+        if: ${{ env.NPM_AGENT_VERSION != env.AGENT_VERSION }}
+        run: |
+          echo "Published agent version (${{env.NPM_AGENT_VERSION }}) is behind local agent version (${{env.AGENT_VERSION}}); exiting."
+          exit 1;
```

Comment:
If it's behind, do we retry later?

Reply:
At this point, we've tried ten times over 20 minutes, so this exits the script with an error. We'd be able to re-run manually after we've verified that NPM has been updated.